pkg/registry/core/pod/storage/eviction.go

```diff
@@ -232,18 +239,41 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		return rtStatus, nil
 	}
 
-	// At this point there was either no PDB or we succeeded in decrementing
+	// At this point there was either no PDB or we succeeded in decrementing or
+	// the pod was unready and we have enough healthy replicas
+
+	deletionOptions := originalDeleteOptions.DeepCopy()
+
+	// Set deletionOptions.Preconditions.ResourceVersion to ensure
+	// the pod hasn't been considered ready since we calculated
+	if updateDeletionOptions {
+		setPreconditionsResourceVersion(deletionOptions, &pod.ResourceVersion)
+	}
 
 	// Try the delete
-	_, _, err = r.store.Delete(ctx, eviction.Name, rest.ValidateAllObjectFunc, originalDeleteOptions.DeepCopy())
+	_, _, err = r.store.Delete(ctx, eviction.Name, rest.ValidateAllObjectFunc, deletionOptions.DeepCopy())
 	if err != nil {
+		if errors.IsConflict(err) && updateDeletionOptions &&
```

Comment:
Is this if block tested via a unit test already?  If not it would be good to add one now.

Reply:
All new logic in this PR is covered by unit testing.  Here's the coverage HTML output if you don't want to run coverage locally: https://gist.github.com/michaelgugino/9d20be02b1f0c4810893c6e81963eaf8